### PR TITLE
MULE-15865: JMS Connector deadlock with ActiveMQ broker reconnection, and forever retry policy

### PR DESCRIPTION
**Pull request by plbalbi**

### CODE REVIEW

File: transports/jms/src/main/java/org/mule/transport/jms/MultiConsumerJmsMessageReceiver.java [hunk 1]
```diff
@@ -152,13 +155,34 @@ public void doWork(RetryContext context) throws Exception
             {
                 try
                 {
+                    JmsConnector connector = (JmsConnector) MultiConsumerJmsMessageReceiver.this.connector;
+                    if (connector.shouldRetryBrokerConnection())
+                    {
+                        try
```
Comment (fsgonz) [inline transports/jms/src/main/java/org/mule/transport/jms/MultiConsumerJmsMessageReceiver.java]: Encapsulate this silent close in a method.

File: transports/jms/src/test/resources/jms-connection-reconnection-with-broker-reconnection.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:vm="http://www.mulesoft.org/schema/mule/vm"
```
Comment (fsgonz) [inline transports/jms/src/test/resources/jms-connection-reconnection-with-broker-reconnection.xml]: Remove unused schemas.

File: transports/jms/src/test/java/org/mule/transport/jms/BrokerReconnectionWithNonExistingQueuesTestCase.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    protected void startBrokerAndAdminSession() throws Exception
+    {
+        broker.start(true);
+        broker.waitUntilStarted();
+
+        connection = new ActiveMQConnectionFactory(this.url).createQueueConnection("admin", "admin");
+        connection.start();
+
+        // Create admin session for further queue creation and connection testing
+        adminSession = (ActiveMQSession) connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
```
Comment (fsgonz) [inline transports/jms/src/test/java/org/mule/transport/jms/BrokerReconnectionWithNonExistingQueuesTestCase.java]: static import.

File: transports/jms/src/test/resources/jms-connection-reconnection-with-broker-reconnection.xml (hunk 2 above)
Comment (plbalbi) [reply transports/jms/src/test/resources/jms-connection-reconnection-with-broker-reconnection.xml]: All imports are used in the config.

[the remark this answers is not in the document]
Comment (plbalbi) [reply transports/jms/src/test/resources/jms-connection-reconnection-with-broker-reconnection.xml]: All imports are used in the config.

Comment (fsgonz) [reply transports/jms/src/test/resources/jms-connection-reconnection-with-broker-reconnection.xml]: You're right. I apologize.

**Pull request merged**